forge-game/src/main/java/forge/game/GameAction.java

```diff
@@ -2608,12 +2596,13 @@ private static void unanimateOnAbortedChange(final SpellAbility cause, final Car
         }
     }
 
-    private CardCollectionView getLastState(final AbilityKey key, final SpellAbility cause, final Map<AbilityKey, Object> params) {
+    public CardCollectionView getLastState(final AbilityKey key, final SpellAbility cause, final Map<AbilityKey, Object> params, final boolean refreshIfEmpty) {
         CardCollectionView lastState = null;
         if (params != null) {
             lastState = (CardCollectionView) params.get(key);
         }
         if (lastState == null && cause != null) {
+            // inside RE
             if (key == AbilityKey.LastStateBattlefield) {
                 lastState = cause.getLastStateBattlefield();
```

Comment:
I'm still not totally convinced this should still be included here:
like if a RE does:
- move X from A to B
- and then **after** moves Y

the second part should still have a newer lastState from its perspective 🤔 

But I would like to find an example (ideally one where it matters) before I commit further

Reply:
Hmn, I checked **Dredge** and it wasn't a problem since it's only set for the root anyway
Will probably leave it alone for now...